### ISSUE TITLE
Dep ensure Solving Failure: No versions of docker.io/go-docker met constraints

### ISSUE THREAD

**Issue by eshnil2000** [play-with-docker/play-with-docker#254]
### Setup: AWS Ubuntu 16.04, ports open 80,443, 3000.

Go installation: /usr/local/go/bin/go
go version go1.10.1 linux/amd64
dep installation: /home/ubuntu/go/bin/dep

PWD installation: /home/ubuntu/go/src/github.com/play-with-docker/play-with-docker

Modified Gopkg.toml added 1 line:

required = ["github.com/googollee/go-socket.io","github.com/urfave/negroni","github.com/docker/distribution","github.com/zbindenren/negroni-prometheus"]

Go Dep solving failure
~/go/bin/dep ensure
Solving failure: No versions of docker.io/go-docker met constraints:
	v1.0.0: Could not introduce docker.io/go-docker@v1.0.0, as it has a dependency on github.com/docker/distribution with constraint edc3ab29cdff8694dd6feb85cfeb4b5f1b38ed9c, which has no overlap with existing constraint ^2.6.0 from (root)
	master: Could not introduce docker.io/go-docker@master, as it has a dependency on github.com/docker/distribution with constraint edc3ab29cdff8694dd6feb85cfeb4b5f1b38ed9c, which has no overlap with existing constraint ^2.6.0 from (root)

Error Details when running docker-compose up:

~/go/src/github.com/play-with-docker/play-with-docker$ docker-compose up
WARNING: The Docker Engine you're using is running in swarm mode.
Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.
To deploy your application across the swarm, use the bundle feature of the Docker experimental build.

Starting haproxy
Starting l2
Starting pwd
Attaching to haproxy, pwd, l2
haproxy    | [WARNING] 108/151956 (1) : config : missing timeouts for frontend 'http-in'.
haproxy    |    | While not properly invalid, you will certainly encounter various problems
haproxy    |    | with such a configuration. To fix this, please ensure that all following
haproxy    |    | timeouts are set to a non-zero value: 'client', 'connect', 'server'.
haproxy    | [WARNING] 108/151956 (1) : config : missing timeouts for backend 'pwd'.
haproxy    |    | While not properly invalid, you will certainly encounter various problems
haproxy    |    | with such a configuration. To fix this, please ensure that all following
haproxy    |    | timeouts are set to a non-zero value: 'client', 'connect', 'server'.
haproxy    | [WARNING] 108/151956 (1) : config : missing timeouts for backend 'l2'.
haproxy    |    | While not properly invalid, you will certainly encounter various problems
haproxy    |    | with such a configuration. To fix this, please ensure that all following
haproxy    |    | timeouts are set to a non-zero value: 'client', 'connect', 'server'.
l2         | 2018/04/19 15:19:57 Connected to network [playwithdocker_default] with ip [172.18.0.3]
pwd        | 2018/04/19 15:19:59 Updating playgrounds configuration
pwd        | 2018/04/19 15:19:59 EVENT: Playground New 4a24b940-e23f-54f5-8867-137433240dd7
pwd        | 2018/04/19 15:19:59 Updating playgrounds configuration
pwd        | 2018/04/19 15:19:59 Listening on port 3000
pwd        | 2018/04/19 15:20:24 Error retrieving playground 8f6492ad-2254-548a-b584-e92a74f54976. Got: NotFound
pwd        | 2018/04/19 15:20:24 Playground for domain 52.39.16.109 was not found!
pwd        | [negroni] Started GET /
pwd        | [negroni] Completed 404 Not Found in 141.916µs

**Comment by eshnil2000** [comment]
I looked up https://github.com/docker/go-docker/blob/master/Gopkg.toml.

I got rid of the error by changing Gopkg.toml:
[[constraint]]
  name = "github.com/docker/distribution"
  revision = "edc3ab29cdff8694dd6feb85cfeb4b5f1b38ed9c"

I ran dep ensure
I also did a go build, go install

I'm not sure if this pulls all the right images, but it got rid of the Solving Error.

Now, I can curl localhost:
curl localhost/my/playground
Gives me: 
{"id":"4a24b940-e23f-54f5-8867-137433240dd7","domain":"localhost","default_dind_instance_image":"franela/dind","available_dind_instance_images":["franela/dind"],"allow_windows_instances":false,"default_session_duration":14400000000000}

but when I curl or try to reach my AWS instance from a browser:
Error retrieving playground d85b75b8-4590-5b9a-8ecf-22fa485e8b34. Got: NotFound
pwd        | 2018/04/20 01:56:07 Playground for domain xxx.us-west-2.compute.amazonaws.com was not found!
